server/controllers/manage/outOfServiceBedCancellationController.ts

```diff
@@ -0,0 +1,51 @@
+import type { Request, RequestHandler, Response } from 'express'
+import { OutOfServiceBedService } from '../../services'
+
+import { catchValidationErrorOrPropogate, fetchErrorsAndUserInput } from '../../utils/validation'
+import paths from '../../paths/manage'
+
+export default class OutOfServiceBedCancellationController {
+  constructor(private readonly outOfServiceBedService: OutOfServiceBedService) {}
+
+  new(): RequestHandler {
+    return async (req: Request, res: Response) => {
+      const { premisesId, bedId, id } = req.params
+      const { errors, errorSummary, errorTitle } = fetchErrorsAndUserInput(req)
+
+      const outOfServiceBed = await this.outOfServiceBedService.getOutOfServiceBed(req.user.token, premisesId, id)
+
+      res.render('manage/outOfServiceBeds/cancel', {
+        pageHeading: 'Cancel out of service bed',
+        errors,
+        errorSummary,
+        errorTitle,
+        outOfServiceBed,
+        backLink: paths.outOfServiceBeds.show({ premisesId, bedId, id, tab: 'details' }),
+        selfLink: paths.outOfServiceBeds.cancel({ premisesId, bedId, id }),
```

Comment:
I believe we've used `submitLink` in another place -- might be good to keep things consistent.

Reply:
OK, yes, there's just one usage in match - but let's stick with that.